source/adios2/helper/adiosNetwork.cpp

```diff
@@ -257,7 +257,10 @@ void HandshakeWriter(Comm const &comm, size_t &appID, std::vector<std::string> &
     std::string localAddressesStr = localAddressesJson.dump();
     std::vector<char> localAddressesChar(64 * channelsPerRank, '\0');
     std::memcpy(localAddressesChar.data(), localAddressesStr.c_str(), localAddressesStr.size());
-    std::vector<char> globalAddressesChar(64 * channelsPerRank * mpiSize, '\0');
+    std::vector<char> globalAddressesChar(static_cast<size_t>(64) *
+                                              static_cast<size_t>(channelsPerRank) *
+                                              static_cast<size_t>(mpiSize),
+                                          '\0');
```

Comment:
Casting the first operand should be enough, you don't need to cast every one.
```suggestion
   std::vector<char> globalAddressesChar(static_cast<size_t>(64) * channelsPerRank * mpiSize, '\0');
```